### ISSUE TITLE
Argument exception on basic example with SNOPT

### ISSUE THREAD

**Issue by DGCaprace** [mdolab/pyoptsparse#174]
## Description
Hi, I'm trying to set up my environment to have pyOptSparse working with SNOPT 7.7. I followed the documentation for the installation, but I get an argument exception when I run one of the basic examples. There might be something I miss but I can't see what. 

### Steps to reproduce issue

1. `$ cd /path/to/pyoptsparse`
2. `$ cp /path/to/SNOPT/src/* pyoptsparse/pySNOPT/source/`
3. `$ rm pyoptsparse/pySNOPT/source/snopth.f`
4. `$ python setup.py build`
5. `$ pip install .`
6. `$ python test/test_hs015.py`

### Current behavior
Here's the stack trace it produces:

```code
======================================================================
ERROR: test_snopt (__main__.TestHS15)
----------------------------------------------------------------------
ValueError: unexpected array size: new_size=1, got array with arr_size=0


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "test/test_hs015.py", line 200, in test_snopt
    self.optimize_with_hotstart("SNOPT", 1e-12, optOptions=optOptions)
  File "test/test_hs015.py", line 165, in optimize_with_hotstart
    self.optimize(optName, tol, storeHistory=True, optOptions=optOptions)
  File "test/test_hs015.py", line 86, in optimize
    sol = opt(optProb, sens=self.sens, storeHistory=self.histFileName, hotStart=hotStart)
  File "/usr/local/lib/python3.8/site-packages/pyoptsparse/pySNOPT/pySNOPT.py", line 487, in __call__
    snopt.snkerc(start, nnCon, nnObj, nnJac, iObj, ObjAdd, ProbNm,
ValueError: failed in converting 27th argument `jcol' of snopt.snlog to C/Fortran array
```

### Code versions
- pyOptSparse v2.2.0
- Python 3.8.5
- SNOPT 7.7 **.3**
- gfortran 5 or 10

**Comment by marcomangano** [comment]
Hi @DGCaprace ! I am not sure if this is going to fix your issue, but step 4 is not required. Once you copied the SNOPT source files in the correct location, just `cd` to the root folder and run `pip install .` (or `pip install . --user` if you are using the system python). 

Moreover, to use the entire suite of tests you can use `testflo` later (see [here](https://mdolab-pyoptsparse.readthedocs-hosted.com/en/latest/install.html#testing))

**Comment by nwu63** [comment]
What is the exact version of SNOPT you are using? I know SNOPT changed the function signature of `snlog` recently. For example I think 7.7.1 works just fine but 7.7.3 does not.

**Comment by DGCaprace** [comment]
Thanks for your quick answers @marcomangano and @nwu63. I am indeed running v7.7.3. I'll see if I can get the 7.7.1

**Comment by nwu63** [comment]
Yeah this is something to bring up with the SNOPT developers, they basically made a backwards-incompatible change to the code without incrementing at least the minor version, and there's very little we can do on our end. Maybe we can talk offline about how to do that if you do plan to reach out to them.

I'm against supporting multiple versions because `f2py` would be pretty clunky, so for now we support the version used internally in the lab, which is v7.7.1.

**Comment by gnowzil** [comment]
Hi!  I'm one of the SNOPT developers.  I recently reverted the changes that caused the incompatibility issue with pyoptsparse.  We've just released the new libraries (snopt 7.7.5).  New library appears to be working with pyoptsparse.
